default/python/AI/pathfinding.py

```diff
@@ -58,32 +193,20 @@ def find_path_with_resupply(start, target, fleet_id, minimum_fuel_at_target=0):
         error("Requested negative fuel at target.")
         return None
 
+    # make sure the minimum fuel at target is realistic
+    if max_fuel < minimum_fuel_at_target:
+        return None
+
     # make sure the target is connected to the start system
     shortest_possible_path_distance = universe.shortestPathDistance(start, target)
     if shortest_possible_path_distance == -1:
         warn("Requested path between disconnected systems, doing nothing.")
         return None
 
-    # make sure the minimum fuel at target is realistic
-    fleet = universe.getFleet(fleet_id)
-    if fleet.maxFuel < minimum_fuel_at_target:
-        return None
-
-    supplied_systems = set(fo.getEmpire().fleetSupplyableSystemIDs)
-    start_fuel = fleet.maxFuel if start in supplied_systems else fleet.fuel
-
-    # We have 1 free jump from supplied system into unsupplied systems.
-    # Thus, the target system must be at most maxFuel + 1 jumps away
-    # in order to reach the system under standard conditions.
-    # In some edge cases, we may have more supply here than what the
-    # supply graph suggests. For example, we could have recently refueled
-    # in a system that is now blockaded by an enemy or we have found
-    # the refuel special.
-    target_distance_from_supply = -min(state.get_system_supply(target), 0)
-    if (fleet.maxFuel + 1 < (target_distance_from_supply + minimum_fuel_at_target) and
-            universe.jumpDistance(start, target) > (start_fuel - minimum_fuel_at_target)):
-        # can't possibly reach this system with the required fuel
-        return None
+    if may_travel_system_func is None:
+        may_travel_system_func = _always_ok_travel_system_func
+    if may_travel_starlane_func is None:
+        may_travel_starlane_func = _always_ok_travel_starlane_func
```

Comment:
Why not use these functions directly as default parameter?

Reply:
Well, if we want to keep the default choices in ```find_path_with_resupply_generic```, which it seems to me we should, then it is simpler for ```find_path_with_resupply``` to be able to pass in ```None``` to use the defaults rather than having to have a complex set of 4 different calling statements into ```find_path_with_resupply_generic``` depending on which, if any, of the defaults it wanted to use.

Reply:
I am not sure what you are refering to. For clarity, I vote to replace
```
def foo(f=None):
if f is None:
    f = some_func
```
with
```
def foo(f=some_func)
```
which should be equivalent except that the second one is shorter and its behavior is obvious from context.